### PR TITLE
Support Python 3.11

### PR DESCRIPTION
**Pull request by phschiele**
## Description
Python 3.11 brings significant speed improvements, better exception messages, and more. It's reasonable to expect that the adoption rate will pick up soon. This PR sets up the CI from our side. 
Still, most solvers do not yet support 3.11.

This PR is meant to collect the availability of dependencies.
- [x] NumPy (starting with 1.23 on PyPI)
- [x] SciPy (starting with 1.9.2 on PyPI)
- [x] SCS (through conda)
- [x] ECOS (through conda)
- [x] OSQP

Other solvers:
- [x] Gurobi
- [x] Clarabel
- [ ] Mosek
TODO: check solvers

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [ ] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

**Comment by michael-123123** [comment]
hi - any timeline on when/if python 3.11 support will be merged into master?

File: sonar-project.properties [hunk 1]
```diff
@@ -4,7 +4,7 @@ sonar.organization=cvxpy
 # Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
 sonar.sources=cvxpy
 sonar.language=py
-sonar.python.version=3.6, 3.7, 3.8, 3.9, 3.10
+sonar.python.version=3.6, 3.7, 3.8, 3.9, 3.10, 3.11
```
Comment (SteveDiamond) [inline sonar-project.properties:7]: We can drop 3.6

**Review by SteveDiamond — APPROVED** [review]

File: cvxpy/tests/test_dgp2dcp.py [hunk 2]
```diff
@@ -328,9 +327,9 @@ def test_solving_non_dcp_problem_raises_error(self) -> None:
         problem = cvxpy.Problem(
           cvxpy.Minimize(cvxpy.Variable(pos=True) * cvxpy.Variable(pos=True)),
         )
-        with self.assertRaisesRegex(error.DCPError,
-                                    r"Problem does not follow DCP "
-                                    "rules(?s)*.*However, the problem does follow DGP rules.*"):
+        with pytest.raises(error.DCPError, match='However, the problem does follow DGP rules'):
+            problem.solve(SOLVER, gp=True)
```
Comment (phschiele) [inline cvxpy/tests/test_dgp2dcp.py:331]: These regexps could no longer be parsed, so I simplified the match statement while keeping the essence of the tests ("does follow DGP/DCP rules").

File: sonar-project.properties (hunk 1 above)
Comment (phschiele) [inline sonar-project.properties:7]:
```suggestion
sonar.python.version=3.7, 3.8, 3.9, 3.10, 3.11
```

Comment (phschiele) [reply sonar-project.properties]: Done.

**Pull request merged**